src/octopeer-github/main/DatabaseAdapter.ts

```diff
@@ -0,0 +1,91 @@
+class DatabaseAdapter {
+
+    private _url: string;
+    private _session: number;
+    private _pr: number;
+    private _user: number;
+    private _initialized: boolean;
+    private _debug: boolean;
+
+    constructor(url: string, debug = false) {
+        const self = this; // scope 'self' to be the adapter (instead of the AJAX call)
+        self._debug = debug;
```

Comment:
What do you mean by "optional constructor"? It is already optional given the default listed in the constructor: `debug = false`.